contrib/mypy/src/python/pants/contrib/mypy/tasks/mypy_task.py

```diff
@@ -29,6 +29,7 @@ class MypyTask(PythonTask):
   @classmethod
   def register_options(cls, register):
     register('--mypy-version', default='0.550', help='The version of mypy to use')
```

Comment:
Both of these options should be marked `fingerprint=True`, to cause changing them to invalidate the build.

Also, you should set `type=file_option` for the config-file, to cause deep fingerprinting of the file.

Reply:
I can do that, although currently the mypy task does not check the invalidation status of the set of targets on which it is operating. I assume that needs to be done as well?

Reply:
Mm. Can ignore me for now then.